grakel_replace/torch_wl_kernel.py

```diff
@@ -1,13 +1,122 @@
 from __future__ import annotations
 
 from collections import Counter
+from typing import Any
 
 import networkx as nx
 import torch
+from botorch.models.gp_regression_mixed import Kernel
 from torch import nn
 
 
-class TorchWLKernel(nn.Module):
+class TorchWLKernel(Kernel):
+    has_lengthscale = False
+
+    def __init__(
+        self,
+        graph_lookup: list[nx.Graph],
+        n_iter: int = 5,
+        *,
+        normalize: bool = True,
+        active_dims: tuple[int, ...],
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(active_dims=active_dims, **kwargs)
+        self.graph_lookup = graph_lookup
+        self.n_iter = n_iter
+        self.normalize = normalize
+
+        # NOTE: set in the `super().__init__()`
+        self.active_dims: torch.Tensor
+
+    def set_graph_lookup(self, graph_lookup: list[nx.Graph]) -> None:
+        self.graph_lookup = graph_lookup
+
+    def forward(
+        self,
+        x1: torch.Tensor,
+        x2: torch.Tensor,
+        *,
+        diag: bool = False,
+        last_dim_is_batch: bool = False,
+        **params: Any,
+    ):
+        if last_dim_is_batch:
+            raise NotImplementedError("TODO: Figure this out")
```

Comment:
Yeah I dunno how to trigger this to be `True` so for now we just leave it as this.